### PR TITLE
Display APM setup task on the WCPay Overview page

### LINKED ISSUE #2169: UPE payment methods task - placeholder
## Description

Display a task in the WCPay home, as well as WC Home to enable additional payment methods.

This ticket's purpose is _only_ to display the task with the correct criteria. Clicking the task can display some placeholder text for the UI work.

See `accountOverviewTaskList`/`_wcpay_feature_account_overview_task_list` feature flag for more details about the existing task list in the WCPay home - p1623726800065300-slack-C01RTUVR81K.

Screen Shot 2021-06-14 at 7 37 38 PM

## Acceptance criteria
- ~Task is displayed when https://github.com/Automattic/woocommerce-payments/issues/2154 is enabled~ - DONE IN https://github.com/Automattic/woocommerce-payments/issues/2178
- ~Task is displayed on WC Home~ - DONE IN https://github.com/Automattic/woocommerce-payments/issues/2178
- Task is displayed in WCPay
- ~Task is "completed" when the merchant has interacted with it and/or added the UPE payment methods~ - DONE IN https://github.com/Automattic/woocommerce-payments/issues/2178
  - ~Task is **not** "completed" when UPE is disabled~
  - ~Task is **not** "completed" when UPE is enabled and the merchant has not interacted with it nor added the UPE payment methods~
- ~Clicking the task shows some placeholder text~ - DONE IN https://github.com/Automattic/woocommerce-payments/issues/2178

### PR DESCRIPTION
**Pull request by waclawjacek**
Fixes #2169

## Changes proposed in this Pull Request

Display the "Boost your sales by accepting new payment methods/Set up additional payment methods" task on the WCPay Overview page (`Payments > Overview`).

(The conditions for displaying the task described below are the same as on `WooCommerce > Home`.)

Either task is displayed only if the `_wcpay_feature_upe_additional_payment_methods` feature flag is enabled. Then:

* If `_wcpay_feature_upe_settings_preview` is enabled, the "Boost your sales" task will be displayed.
  * Marked as complete if either you go through the setup or `_wcpay_feature_upe` is enabled.
* If `_wcpay_feature_upe_settings_preview` is disabled and `_wcpay_feature_upe` is enabled, the "Set up additional methods" task will be displayed.
  * Marked as complete only if you go through the setup.

## Known issues

I was able to resolve an issue with the buttons not working (mentioned in the "Implementation details" section) and create a follow-up issue for removing the button thanks to information from @octaedro (p1625663464222300-slack-C9K5T0XSP). Thank you!

* The task list item design displays a button (which seems to be inconsistent with the desired design in the issue #2169).
  * This was implemented in https://github.com/woocommerce/woocommerce-admin/pull/7263 but doesn't yet seem to be available in WCPay.
  * Created follow-up issue: #2455
* To display the task, the code redirects to the task's WC > Home page. This has the consequence that clicking the "Back" button in the top-left corner of the page results in going to WC > Home even if you initiate the wizard on WCPay > Overview.
  * Created follow-up issue: #2454

This GIF demonstrates the issues plus it also demonstrates that the only clickable task component is the button: https://d.pr/i/0zS9C2

## Testing instructions

1. Clear the DB to make sure the task is not marked as completed:
  ```sql
  DELETE FROM `wp_options` WHERE `option_name` = 'wcpay_additional_methods_setup_completed'
  ```
2. Verify that:
  1. Tasks are displayed as described in the "Changes proposed in this Pull Request" section.
  2. Clicking the button on the WCPay > Overview page redirects to the task.
  3. Clicking the button on the WC > Home page was left intact.

## Screenshots

(The other task is just to illustrate the button always being displayed.)

### "Boost your sales" task (UPE preview enabled)

Pending task: 

Markup on 2021-07-06 at 12:17:09

Completed task:

Markup on 2021-07-06 at 12:20:06

### "Set up additional methods" task (UPE preview disabled, UPE enabled)

Pending task:

Markup on 2021-07-06 at 12:23:58

Completed task:

Markup on 2021-07-06 at 12:23:12

## Implementation details

Since WCPay already has a mechanism for displaying a task list shared between `WC > Home` and `WCPay > Overview`, I have refactored the existing code used to create it to make use of this mechanism, moving the logic from `WC_Payments_Admin_Additional_Methods_Setup` to `WC_Payments_Admin` and removing the former class.

I have also refactored the existing task to use callbacks provided by a new context rather than setting `window.*` variables directly.

This PR also [contains a fix](https://github.com/Automattic/woocommerce-payments/pull/2358/files#diff-64da8dc3c200ed8ee47daef3203920aa6d1554e509a9eda184c959844b90991bR87) for a bug where, if meeting the conditions for the "Boost your sales" task to be displayed, the task was not marked as completed if UPE was enabled but you didn't go through the stepper. This was because we were checking for `'1' === isUpeEnabled` but the value returned from the back-end was boolean.

After a button on the WCPay > Overview task list was clicked (including for existing tasks), an error was thrown in the developer console and nothing else happened. In #2365, tasks' `onClick` props are mapped to `action` props supported by `woocommerce-admin`'s `<TaskList>`. I have [replicated this in this PR](https://github.com/Automattic/woocommerce-payments/pull/2358/files#diff-34ab3d28feff22ba3c825956998f112205530458a81a255ebfdfc293d8221269R46).

-------------------

- [x] Added changelog entry (or does not apply)
  - Not added as this is a part of a larger feature.
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

File: includes/admin/class-wc-payments-admin.php [hunk 1]
```diff
@@ -254,25 +254,31 @@ public function register_payments_scripts() {
...
-			'wpcomReconnectUrl'     => $this->payments_api_client->is_server_connected() && ! $this->payments_api_client->has_server_connection_owner() ? WC_Payments_Account::get_wpcom_reconnect_url() : null,
+			'zeroDecimalCurrencies'  => WC_Payments_Utils::zero_decimal_currencies(),
+			'fraudServices'          => $this->account->get_fraud_services_config(),
+			'isJetpackConnected'     => $this->payments_api_client->is_server_connected(),
+			'accountStatus'          => $this->account->get_account_status_data(),
+			'accountFees'            => $this->account->get_fees(),
+			'showUpdateDetailsTask'  => get_option( 'wcpay_show_update_business_details_task', 'no' ),
+			'wpcomReconnectUrl'      => $this->payments_api_client->is_server_connected() && ! $this->payments_api_client->has_server_connection_owner() ? WC_Payments_Account::get_wpcom_reconnect_url() : null,
+			'additionalMethodsSetup' => [
+				'showTask'                    => $this->is_page_eligible_for_additional_methods_setup_task(),
```
Comment (frosso) [inline includes/admin/class-wc-payments-admin.php:277]: The other attributes are using `is`/`has` prefixes - what do you think about doing the same for this? (e.g.: `isTaskVisible`)

**Review by frosso — APPROVED** [review]
Thanks for the thorough overview!

I have one minor comment about the `showTask` attribute. The changes look good and seem accurate 👍 as always, I like the thoroughness of the tests.

The added `AdditionalMethodsSetupContext` context provider seems maybe less necessary and would have probably helped reduce the number of file changes and the number of arguments on `createAdditionalMethodsSetupTask`.
After all, the global variables are there only to mark the task as completed in the list, and are not used by determine the rest of the application's state, similar to this: https://github.com/woocommerce/woocommerce-admin/blob/main/docs/examples/extensions/add-task/js/index.js#L24

:shipit:

File: includes/admin/class-wc-payments-admin.php (hunk 1 above)
Comment (waclawjacek) [reply includes/admin/class-wc-payments-admin.php:277]: Yes! 💯

**Comment by waclawjacek** [comment]
Thanks so much for the review and the kind words!

> The added `AdditionalMethodsSetupContext` context provider seems maybe less necessary and would have probably helped reduce the number of file changes and the number of arguments on `createAdditionalMethodsSetupTask`.
> After all, the global variables are there only to mark the task as completed in the list, and are not used by determine the rest of the application's state, similar to this: https://github.com/woocommerce/woocommerce-admin/blob/main/docs/examples/extensions/add-task/js/index.js#L24

You're right! I have included both your suggestions in the PR. Could you please give the updates a look?

**Review by frosso — APPROVED** [review]
Thanks for the changes! Still looks good, tested the scenarios locally 👍 :shipit:

**Pull request merged**